src/fairchem/core/units/mlip_unit/mlip_unit.py

```
@@ -965,6 +967,47 @@ def get_composition_charge_spin_dataset(self, data):
         )
         return comp_charge_spin, getattr(data, "dataset", [None])
 
+    def populate_empty_prediction(self) -> dict:
+        """
+        Populate preduction dict with zeroes for all values (with number of atoms = 1)
+        """
+        pred_output = {}
+        for task_name, task in self.tasks.items():
+            if task.property == "energy":
+                pred_output[task_name] = torch.Tensor([0.0])
+            elif task.property == "forces":
+                pred_output[task_name] = torch.Tensor([[0.0] * 3])
+            elif task.property == "stress":
+                pred_output[task_name] = torch.Tensor([[0.0] * 9])
+        return pred_output
+
+    def get_single_atom_energies(self, data) -> dict:
+        """
+        Populate output with single atom energies
+        """
+        if self.atom_refs is None:
+            raise ValueError(
+                "Single atom system but no atomic references present. "
+                "Please call fairchem.core.pretrained_mlip.get_predict_unit() "
+                "with an appropriate checkpoint name."
+            )
+        if data.charge.item() != 0:
+            raise ValueError(
+                "This model cannot handle single atom systems with non-zero charge."
+            )
+        logging.warning(
+            "Single atom systems are not handled by the model; "
+            "the precomputed DFT result is returned. "
+            "Spin multiplicity is ignored for monoatomic systems."
+        )
+        elt = data.atomic_numbers.item()
+        pred_output = self.populate_empty_prediction()
+        for task_name, task in self.tasks.items():
+            if task.property == "energy":
+                atom_refs = self.atom_refs[task_name.replace("_energy", "_elem_refs")]
+                pred_output[task_name] = torch.Tensor([atom_refs.get(elt, 0.0)])
```

Are we concerned someone will use this for a non-supported element, get 0 and be confused? Better to just throw an error?

---

Part of the problem is that the current paradigm populates every task energy regardless of what task the user requested. So if they ask for the OMol head and we were missing that element from say OC20, if we raise an error we will crash doing something the user doesn't care about.

---

maybe iterate over the data points in `data` and check that data point's `dataset` value and raise an Error  if the element reference for that dataset does not exist?

---

So this dataset value might be "OC20" or "OMol"? Why is this not in the `self.tasks`?

---

yup, the `dataset` attribute of a data point should be "oc20" or "omol".

Tasks are a model level attribute. Each data point in a batch has a `dataset` attribute.